apollo-federation/tests/subgraph/subgraph_validation_tests.rs

```diff
@@ -0,0 +1,300 @@
+use apollo_federation::error::FederationError;
+use apollo_federation::subgraph::SubgraphError;
+use apollo_federation::subgraph::typestate::Subgraph;
+
+fn build_for_errors(schema: &str, err_reason: &str) -> Result<Vec<Vec<String>>, SubgraphError> {
+    let err = Subgraph::parse("S", "", schema)
+        .expect("parses schema")
+        .expand_links()
+        .expect("expands links")
+        .validate(true)
+        .expect_err(err_reason);
```

Comment:
I don't think we need to take `err_reason` here. It's just a debug message. We can just do `.expect_err("subgraph error was expected")`, since we don't expect to see this string if the test was successful.